test/Kozani/data/KozaniManager.Msix/KozaniManager.Msix.vcxproj

```diff
@@ -0,0 +1,111 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup Label="ProjectConfigurations">
+    <ProjectConfiguration Include="Debug|Win32">
+      <Configuration>Debug</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|Win32">
+      <Configuration>Release</Configuration>
+      <Platform>Win32</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Debug|x64">
+      <Configuration>Debug</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|x64">
+      <Configuration>Release</Configuration>
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
```

Comment:
Did you create the project via VS File/New...?

Just a hunch. Since there's no ARM64. And lines 28+ are highly redundant and not the optimized form we use. And...

Please use the standard project templates per `tools\AddProject -List`. Notably, `test.cpp.dll.taef` or `test.cpp.dll.msixpackage`. Do we need to add C++/WinRT support to `test.cpp.dll.msixpackage` or is there more to it you need?

Reply:
This is copied and modified from existing msxi project (D:\Projects\WindowsAppSDK\test\DynamicDependency\data\DynamicDependency.DataStore.Msix\DynamicDependencies.DataStore.Msix.vcxproj). The current template does not meet the need as here we only need to package existing dll and exe from other projects. No need to add new dll code.

Reply:
>No need to add new dll code.

Then you'll 

Compare with the template `tools\ProjectTemplates\test.cpp.dll.msixpackage\PurojekutoTenpuret.vcxproj` and you'll quickly see how the config differs If you're just packaging and don't need to compile anything into a dll or exe you'll want a new project template (more of a 'utility' than `<ConfigurationType>DynamicLibrary</>`).

Or even simpler, you don't need a new project. Your project producing your test dll can ALSO package it. Go into your `.vcxproj` that makes your test dll and add `makemsix.targets` support e.g. the core of it's

```
  <PropertyGroup>
    <MakeMsixOutputFilename>$(MSBuildProjectName).msix</MakeMsixOutputFilename>
  </PropertyGroup>
  <ItemGroup>
    <MakeMsixInputs Include="$(MSBuildThisFileDirectory)appxmanifest.xml" />
    <MakeMsixInputs Include="$(MSBuildThisFileDirectory)logo.png" />
    <MakeMsixInputs Include="$(OutDir)$(TargetFileName)" />
  </ItemGroup>
  <PropertyGroup>
    <PackageCertificateKeyFile>$(RepoTestCertificatePFX)</PackageCertificateKeyFile>
    <PackageCertificatePassword>$(RepoTestCertificatePassword)</PackageCertificatePassword>
  </PropertyGroup>
  <Import Project="$(RepoRoot)\MakeMsix.targets" />
```

That adds a post-build step so after your project compiles its DLL it then runs commands in MakeMSIX.targets (notably, makeappexe+signtool.exe) to create your .msix.

No need for an "msix-dedicated"(ish) project.

Reply:
I am following existing pattern in WinAppSDK project for "msix-dedicated"(ish) projects, which I found intuitive and convenient for creating test MSIX packages. Those packages can refer to multiple code projects that build various exe and dlls. A dedicated packaging project to organize relevant build outputs makes sense and it is the model followed by product code like MSRDC. They has dedicated MSIX packaging project like this as well. Note that the test MSIX package can be used in different test projects so for a test code project to build it seems less intuitive.